### PR TITLE
"Other methods" TypeScript tests

### PR DESCRIPTION
**Pull request by willsoto**
#1431

### CODE REVIEW

**Comment by koskimas** [comment]
This is failing in CI for real an not because of random fails.

**Comment by willsoto** [comment]
Yes sorry! I’m gonna wrap this one up soon, just been really busy.

File: package.json [hunk 1]
```diff
@@ -112,12 +112,12 @@
     }
   },
   "lint-staged": {
-    "*.{js,ts}": [
+    "*.js": [
```
Comment (willsoto) [inline package.json]: When I originally made this PR, I forgot to test eslint on `ts` files, and it doesn't support `ts` right now, so I've moved them into the other hook.

File: tests/ts/query-builder-api/other-methods.ts [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+  builder.then(
+    () => {
+      console.log('success');
+    },
+    () => {
+      console.log('error');
+    }
+  );
+
+  // builder.map((obj) => obj);
```
Comment (willsoto) [inline tests/ts/query-builder-api/other-methods.ts]: These are not present in the typings currently, but I wonder if they are being deprecated like the other similar methods (`pluck`, etc).

File: tests/ts/query-builder-api/other-methods.ts [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+  // builder.reduce()
+
+  builder.catch(error => {
+    console.log(error);
+  });
+
+  // builder.return()
+  // builder.bind()
+  // builder.asCallback();
+  // builder. nodeify()
```
Comment (willsoto) [inline tests/ts/query-builder-api/other-methods.ts]: No typings for this, unsure if deprecated or removed in v2.

File: tests/ts/query-builder-api/other-methods.ts [hunk 4]
```diff
@@ -0,0 +1,202 @@
...
+    onBuild(builder: any) {}
+  });
+
+  Person.query()
+    .withGraphFetched('[movies, children.movies]')
+    .context({
+      onBuild(builder: any) {
+        builder.withSchema('someSchema');
+      }
+    });
```
Comment (willsoto) [inline tests/ts/query-builder-api/other-methods.ts]: All the various context callbacks are untyped right now, hence the `any`

**Comment by willsoto** [comment]
@koskimas few questions/comments - but otherwise ready for review.

**Comment by willsoto** [comment]
@koskimas Now that 2.0 is released, do you think you have a chance to take a look at this?

**Pull request merged**